lib/vector-core/src/schema/registry.rs

```diff
@@ -0,0 +1,112 @@
+use crate::schema;
+use once_cell::sync::OnceCell;
+use std::{collections::HashMap, fmt, num::NonZeroU16};
+use value::Kind;
+
+static REGISTRY: OnceCell<Registry> = OnceCell::new();
+
+/// Initialize the global schema registry.
+///
+/// TODO(Jean):
+///
+///   It'd be nice if we could thread the registry through the topology to the relevant
+///   sources/sinks, without having to access a global static.
+///
+///   However, given the churn thus causes, and the fact that we already use the same global static
+///   pattern for the (unrelated) `LOG_SCHEMA` (see `vector-core`), this is the easiest solution to
+///   the problem.
+///
+///   We'll still migrate to a non-global solution eventually, just not at the moment.
+///
+/// # Panics
+///
+/// If trying to initialize the registry more than once.
+pub fn init(registry: Registry) {
+    REGISTRY
+        .set(registry)
+        .expect("must be registered exactly once");
+}
+
+/// Get an immutable reference to the global schema registry.
+///
+/// # Panics
+///
+/// If the registry is unitialized (see `init_registry`).
+pub fn global() -> &'static Registry {
+    REGISTRY.get().expect("must call `init_registry` first")
+}
+
+/// A global schema registry that tracks known event schemas and their information.
+///
+/// Each event gets assigned a [`schema::Id`], which corresponds to a schema existing in this
+/// registry.
+#[derive(Debug)]
+pub struct Registry {
+    /// A list of schema definitions.
+    definitions: HashMap<schema::Id, schema::Definition>,
+
+    /// Keep track of the next available schema identifier.
+    ///
+    /// The 0th identifier is reserved for the default "open" schema.
```

Comment:
Small nit to match language used elsewhere
```suggestion
    /// The 0th identifier is reserved for the default "empty" schema.
```